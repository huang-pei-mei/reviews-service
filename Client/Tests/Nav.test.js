import 'jsdom-global/register';
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import "@babel/polyfill";
import Nav from '../src/Components/nav.js';
import mockSuccessAjax from './mockSuccessAjax';
import successData  from './successData.json';
import renderer from 'react-test-renderer'

it ('should match snapshot', () => {
  const state  = {reviews: [], carouselReviews: [], itemsToShow: 10, Audible: 'Audible', Canada: 'Canada', reviewBodyClass: 'defaultReview', readMoreDisplay: 'readMore', hideMeDisplay: 'hideHideMeButton'};

  const tree = renderer.create(<Nav state={state}/>).toJSON();
  expect(tree).toMatchSnapshot();
})

it ('should render audible class correctly', () => {
  const state  = {reviews: [], carouselReviews: [], itemsToShow: 10, Audible: 'Audible', Canada: 'Canada', reviewBodyClass: 'defaultReview', readMoreDisplay: 'readMore', hideMeDisplay: 'hideHideMeButton'};

  const navBody = shallow(<Nav state={state} />);
  expect((navBody.find('.Audible')).exists()).toBe(true);
})