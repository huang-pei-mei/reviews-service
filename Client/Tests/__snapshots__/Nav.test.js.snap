// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<nav>
  <div
    className="formattingDiv"
  />
  <div
    className="greyBar"
  />
  <button
    className="Canada"
    onClick={[Function]}
  >
    Audible.co.ca Reviews
  </button>
  <button
    className="Audible"
    onClick={[Function]}
  >
    Audible.com Reviews
  </button>
  <div
    className="customerReviewsDiv"
  >
    Customer Reviews
  </div>
</nav>
`;
